commandapi-platforms/commandapi-spigot/commandapi-spigot-core/src/main/java/dev/jorel/commandapi/nms/SpigotNMS.java

```diff
@@ -0,0 +1,20 @@
+package dev.jorel.commandapi.nms;
+
+import com.mojang.brigadier.context.CommandContext;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.brigadier.tree.CommandNode;
+import net.md_5.bungee.api.chat.BaseComponent;
+import org.bukkit.ChatColor;
+import org.bukkit.command.Command;
+
+public interface SpigotNMS<CommandListenerWrapper> {
+
+	BaseComponent[] getChat(CommandContext<CommandListenerWrapper> cmdCtx, String key) throws CommandSyntaxException;
+
+	ChatColor getChatColor(CommandContext<CommandListenerWrapper> cmdCtx, String key);
+
+	BaseComponent[] getChatComponent(CommandContext<CommandListenerWrapper> cmdCtx, String key);
+
+	NMS<?> bukkitNMS();
```

Comment:
This (and the same method in PaperNMS) could use the generic parameter to avoid a [cast in `CommandAPISpigot`](https://github.com/JorelAli/CommandAPI/blob/7d2a622ab856eabf2393d8c35bee309bad646177/commandapi-platforms/commandapi-spigot/commandapi-spigot-core/src/main/java/dev/jorel/commandapi/CommandAPISpigot.java#L36).

```suggestion
	NMS<CommandListenerWrapper> bukkitNMS();
```